channeld/channel.c

```diff
@@ -1977,6 +1977,94 @@ static void handle_preimage(struct peer *peer, const u8 *inmsg)
 	abort();
 }
 
+static u8 *make_failmsg(const tal_t *ctx,
+			struct peer *peer,
+			const struct htlc *htlc,
+			enum onion_type failcode)
+{
+	u8 *msg, *channel_update = NULL;
+	u32 cltv_expiry = abs_locktime_to_blocks(&htlc->expiry);
+
+	switch (failcode) {
+	case WIRE_INVALID_REALM:
+		msg = towire_invalid_realm(ctx);
+		goto done;
+	case WIRE_TEMPORARY_NODE_FAILURE:
+		msg = towire_temporary_node_failure(ctx);
+		goto done;
+	case WIRE_PERMANENT_NODE_FAILURE:
+		msg = towire_permanent_node_failure(ctx);
+		goto done;
+	case WIRE_REQUIRED_NODE_FEATURE_MISSING:
+		msg = towire_required_node_feature_missing(ctx);
+		goto done;
+	case WIRE_TEMPORARY_CHANNEL_FAILURE:
+		channel_update = create_channel_update(ctx, peer, true);
```

Comment:
Hm, hopefully this only ever happens when the connection to the peer is lost. This message could spread through the network and disable the channel globally, so we need to re-enable it later.

Reply:
Come to think of it, this is the `channeld` that is taking care of the connection, it'll die when we lose connectivity, so this case will never trigger, unless we are connected and have some other issue (maybe reconnecting). I thought the temporary channel failure was supposed to be a sort of catch-all error, so just disabling this may be dangerous.